src/PeriodEnd/LevyPayments/DeployLevyPayments/sql/dml/05 PeriodEnd.Populate.Reference.PaymentsHistory.dml.sql

```diff
@@ -15,19 +15,29 @@ GO
 
 INSERT INTO [Reference].[LevyPaymentsHistory]
 
-SELECT
-
-	p.RequiredPaymentId,
-	rp.CommitmentId ,
-	p.DeliveryMonth ,
-	p.DeliveryYear ,
-	p.TransactionType ,
-	p.FundingSource,
-	p.Amount 
-    FROM  ${DAS_PeriodEnd.FQ}.Payments.Payments p 
-	JOIN  ${DAS_PeriodEnd.FQ}.PaymentsDue.RequiredPayments rp on p.RequiredPaymentId = rp.Id
-    WHERE rp.Ukprn IN (SELECT DISTINCT [Ukprn] FROM [Reference].[Providers])
-	AND IsNull(rp.CommitmentId,0) > 0
+SELECT p.RequiredPaymentId,
+    p.CommitmentId,
+    p.DeliveryMonth,
+    p.DeliveryYear,
+    p.TransactionType,
+    p.FundingSource,
+    p.Amount
+FROM OPENQUERY(${DAS_PeriodEnd.servername}, '
+		SELECT 
+			p.RequiredPaymentId,
+			rp.Ukprn ,
+			rp.CommitmentId ,
+			p.DeliveryMonth ,
+			p.DeliveryYear ,
+			p.TransactionType ,
+			p.FundingSource,
+			p.Amount 
+		FROM 
+			${DAS_PeriodEnd.databasename}.Payments.Payments p 
+			JOIN ${DAS_PeriodEnd.databasename}.PaymentsDue.RequiredPayments rp on p.RequiredPaymentId = rp.Id'
+    ) AS p
+WHERE p.Ukprn IN (SELECT DISTINCT [Ukprn] FROM [Reference].[Providers])
+    AND IsNull(p.CommitmentId, 0) > 0
```

Comment:
Looks like "IsNull(p.CommitmentId, 0) > 0" could be part of the OPENQUERY where clause, but would need to measure to see if it would be an optimisation.